server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -118,7 +118,12 @@ final class TransportHandshaker {
 
     static final TransportVersion EARLIEST_HANDSHAKE_VERSION = TransportVersion.fromId(6080099);
     static final TransportVersion REQUEST_HANDSHAKE_VERSION = TransportVersions.MINIMUM_COMPATIBLE;
-    static final Set<TransportVersion> ALLOWED_HANDSHAKE_VERSIONS = Set.of(EARLIEST_HANDSHAKE_VERSION, REQUEST_HANDSHAKE_VERSION);
+    static final TransportVersion V9_HANDSHAKE_VERSION = TransportVersion.fromId(8_800_00_0);
```

Comment:
Note that any transport version understood by 8.18.0 and not 8.17.x will do here, it just so happens that there's a numerologically-cute one in that range which I've picked. (In fact it might not even matter whether 8.17.x understands this transport version or not since the handshake path is so different from the regular path, as long as it correctly interprets ints and vints and so on).